codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/RestParserGeneratorBase.java

```diff
@@ -357,17 +349,13 @@ public Void stringShape(StringShape shape) {
 
         @Override
         public Void booleanShape(BooleanShape shape) {
-            splitByComma()
-                    .write(".map { |s| s == 'true' }")
-                    .dedent();
+            writer.write("$1LHearth::Http::HeaderListParser.parse_boolean_list($2L)", dataSetter, valueGetter);
             return null;
         }
 
         @Override
         public Void integerShape(IntegerShape shape) {
-            splitByComma()
-                    .write(".map { |s| s.to_i }")
-                    .dedent();
+            writer.write("$1LHearth::Http::HeaderListParser.parse_integer_list($2L)", dataSetter, valueGetter);
```

Comment:
So I think this was an issue with the previous code likely - but I *think* header lists can contain any number type (from the documentation the trait selector says  `list > member > :test(boolean, number, string, timestamp)))` - so I think the `number` means we need to cover all numeric types here.  We can cover that with two methods, a parse_integer_list and parse_float_list.  But I think we should probably add that.

Reply:
Updated to cover all numeric types!